2024-02-14T05:29:14.7975360Z TRACE:: Config.Get(): Configuration key not found. Key: version-id - Using default value: 
2024-02-14T05:29:14.8463475Z TRACE:: Config.Get(): Configuration key not found. Key: cache-location - Using default value: 
2024-02-14T05:29:14.8470484Z TRACE:: Engine.Main(): LEAN ALGORITHMIC TRADING ENGINE v2.5.0.0 Mode: DEBUG (64bit) Host: LAPTOP-BC2TTKVT
2024-02-14T05:29:14.8538233Z TRACE:: Engine.Main(): Started 5:29 AM
2024-02-14T05:29:14.8635764Z TRACE:: Config.Get(): Configuration key not found. Key: lean-manager-type - Using default value: LocalLeanManager
2024-02-14T05:29:14.9017332Z TRACE:: JobQueue.NextJob(): Selected /LeanCLI/main.py
2024-02-14T05:29:15.0538293Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-capacity - Using default value: 120
2024-02-14T05:29:15.0544468Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-time-interval-minutes - Using default value: 1440
2024-02-14T05:29:15.0549373Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-refill-amount - Using default value: 18
2024-02-14T05:29:15.0555396Z TRACE:: Config.GetValue(): maximum-chart-series - Using default value: 30
2024-02-14T05:29:15.0574308Z TRACE:: Config.GetValue(): storage-limit - Using default value: 10737418240
2024-02-14T05:29:15.0581341Z TRACE:: Config.GetValue(): storage-permissions - Using default value: 3
2024-02-14T05:29:15.0586707Z TRACE:: Config.Get(): Configuration key not found. Key: optimization-id - Using default value: 
2024-02-14T05:29:15.0591599Z TRACE:: Config.Get(): Configuration key not found. Key: backtest-name - Using default value: local
2024-02-14T05:29:15.0728628Z TRACE:: Config.Get(): Configuration key not found. Key: data-channel - Using default value: 
2024-02-14T05:29:15.0733784Z TRACE:: Config.Get(): Configuration key not found. Key: job-organization-id - Using default value: 
2024-02-14T05:29:15.0739530Z TRACE:: Config.Get(): Configuration key not found. Key: python-venv - Using default value: 
2024-02-14T05:29:15.0744597Z TRACE:: Config.Get(): Configuration key not found. Key: out-of-sample-max-end-date - Using default value: 
2024-02-14T05:29:15.0754227Z TRACE:: Config.GetValue(): out-of-sample-days - Using default value: 0
2024-02-14T05:29:15.0783573Z TRACE:: Config.Get(): Configuration key not found. Key: data-permission-manager - Using default value: DataPermissionManager
2024-02-14T05:29:15.1329821Z TRACE:: Config.GetValue(): zip-data-cache-provider - Using default value: 10
2024-02-14T05:29:15.1386201Z TRACE:: Config.Get(): Configuration key not found. Key: fundamental-data-provider - Using default value: CoarseFundamentalDataProvider
2024-02-14T05:29:15.1485561Z TRACE:: AlgorithmManager.CreateTokenBucket(): Initializing LeakyBucket: Capacity: 120 RefillAmount: 18 TimeInterval: 1440
2024-02-14T05:29:15.1499453Z TRACE:: Config.GetValue(): algorithm-manager-time-loop-maximum - Using default value: 20
2024-02-14T05:29:15.1860921Z TRACE:: Engine.Run(): Resource limits '0' CPUs. 2147483647 MB RAM.
2024-02-14T05:29:15.1876051Z TRACE:: TextSubscriptionDataSourceReader.SetCacheSize(): Setting cache size to 71582788 items
2024-02-14T05:29:15.3072924Z TRACE:: Config.GetValue(): downloader-data-update-period - Using default value: 7
2024-02-14T05:29:15.9404235Z TRACE:: DebuggerHelper.Initialize(): initializing python...
2024-02-14T05:29:15.9415405Z TRACE:: PythonInitializer.Initialize(): start...
2024-02-14T05:29:17.8559538Z TRACE:: PythonInitializer.Initialize(): ended
2024-02-14T05:29:17.8565695Z TRACE:: DebuggerHelper.Initialize(): python initialization done
2024-02-14T05:29:17.8570335Z TRACE:: DebuggerHelper.Initialize(): starting...
2024-02-14T05:29:17.8575445Z TRACE:: DebuggerHelper.Initialize(): waiting for PTVSD debugger to attach at localhost:5678...
2024-02-14T05:29:20.5843230Z TRACE:: DebuggerHelper.Initialize(): started
2024-02-14T05:29:20.5852255Z TRACE:: Config.GetValue(): algorithm-creation-timeout - Using default value: 90
2024-02-14T05:29:20.5923951Z TRACE:: AlgorithmPythonWrapper(): Python version 3.8.16 (default, Jan 17 2023, 23:21:03) 
[GCC 11.2.0]: Importing python module main
2024-02-14T05:29:33.1069997Z TRACE:: AlgorithmPythonWrapper(): main successfully imported.
2024-02-14T05:29:33.1139837Z TRACE:: AlgorithmPythonWrapper(): Creating IAlgorithm instance.
2024-02-14T05:29:33.2894558Z TRACE:: LocalObjectStore.Initialize(): Storage Root: /Storage. StorageFileCount 9999999. StorageLimit 10240MB
2024-02-14T05:29:33.2993906Z TRACE:: Config.GetValue(): zip-data-cache-provider - Using default value: 10
2024-02-14T05:29:33.3075516Z TRACE:: HistoryProviderManager.Initialize(): history providers [SubscriptionDataReaderHistoryProvider]
2024-02-14T05:29:33.3124946Z TRACE:: BacktestingSetupHandler.Setup(): Setting up job: UID: 255804, PID: 388644080, Version: 2.5.0.0, Source: WebIDE
2024-02-14T05:29:33.3215223Z TRACE:: Config.Get(): Configuration key not found. Key: security-data-feeds - Using default value: 
2024-02-14T05:29:33.4735156Z TRACE:: Config.GetValue(): data-feed-workers-count - Using default value: 16
2024-02-14T05:29:33.4773509Z TRACE:: Config.GetValue(): data-feed-max-work-weight - Using default value: 400
2024-02-14T05:29:33.4777504Z TRACE:: WeightedWorkScheduler(): will use 16 workers and MaxWorkWeight is 400
2024-02-14T05:29:33.9823059Z TRACE:: BacktestingResultHandler(): Sample Period Set: 104.40
2024-02-14T05:29:33.9871574Z TRACE:: Config.GetValue(): forward-console-messages - Using default value: True
2024-02-14T05:29:34.1813259Z ERROR:: Engine.Run(): During the algorithm initialization, the following exception has occurred: module 'tensorflow' has no attribute 'get_default_graph'
  at Initialize
    self.graph = tf.get_default_graph()
   at Python.Runtime.PythonException.ThrowLastAsClrException()
   at Python.Runtime.PyObject.Invoke(PyTuple args in main.py: line 232
 module 'tensorflow' has no attribute 'get_default_graph'
2024-02-14T05:29:34.1824041Z TRACE:: JOB HANDLERS:
         DataFeed:             QuantConnect.Lean.Engine.DataFeeds.FileSystemDataFeed
         Setup:                QuantConnect.Lean.Engine.Setup.BacktestingSetupHandler
         RealTime:             QuantConnect.Lean.Engine.RealTime.BacktestingRealTimeHandler
         Results:              QuantConnect.Lean.Engine.Results.BacktestingResultHandler
         Transactions:         QuantConnect.Lean.Engine.TransactionHandlers.BacktestingTransactionHandler
         Object Store:         QuantConnect.Lean.Engine.Storage.LocalObjectStore
         History Provider:     QuantConnect.Lean.Engine.HistoricalData.HistoryProviderManager
         Brokerage:            QuantConnect.Brokerages.Backtesting.BacktestingBrokerage
         Data Provider:        QuantConnect.Lean.Engine.DataFeeds.DefaultDataProvider

2024-02-14T05:29:34.1831820Z TRACE:: FileSystemDataFeed.Exit(): Start. Setting cancellation token...
2024-02-14T05:29:34.1852939Z TRACE:: FileSystemDataFeed.Exit(): Exit Finished.
2024-02-14T05:29:34.1860247Z TRACE:: BacktestingResultHandler.Exit(): starting...
2024-02-14T05:29:34.2087967Z TRACE:: BacktestingResultHandler.Exit(): Saving logs...
2024-02-14T05:29:34.2146329Z ERROR:: During the algorithm initialization, the following exception has occurred: module 'tensorflow' has no attribute 'get_default_graph'
  at Initialize
    self.graph = tf.get_default_graph()
   at Python.Runtime.PythonException.ThrowLastAsClrException()
   at Python.Runtime.PyObject.Invoke(PyTuple args in main.py: line 232
 module 'tensorflow' has no attribute 'get_default_graph'
 module 'tensorflow' has no attribute 'get_default_graph'
  at Initialize
    self.graph = tf.get_default_graph()
   at Python.Runtime.PythonException.ThrowLastAsClrException()
   at Python.Runtime.PyObject.Invoke(PyTuple args in main.py: line 232
 module 'tensorflow' has no attribute 'get_default_graph'
2024-02-14T05:29:34.2148406Z TRACE:: StopSafely(): Waiting for 'Result Thread' thread to stop...
2024-02-14T05:29:34.2150816Z TRACE:: Debug: 123
2024-02-14T05:29:34.2155859Z TRACE:: Debug: Your log was successfully created and can be retrieved from: /Results/1428326774-log.txt
2024-02-14T05:29:34.2160275Z TRACE:: BacktestingResultHandler.Run(): Ending Thread...
2024-02-14T05:29:34.3471833Z TRACE:: 
STATISTICS:: OrderListHash d41d8cd98f00b204e9800998ecf8427e
2024-02-14T05:29:34.3476274Z TRACE:: BacktestingResultHandler.SendAnalysisResult(): Processed final packet
2024-02-14T05:29:34.3486709Z TRACE:: Engine.Run(): Disconnecting from brokerage...
2024-02-14T05:29:34.3492459Z TRACE:: Engine.Run(): Disposing of setup handler...
2024-02-14T05:29:34.3496374Z TRACE:: Engine.Main(): Analysis Completed and Results Posted.
2024-02-14T05:29:34.3509489Z TRACE:: StopSafely(): Waiting for '' thread to stop...
2024-02-14T05:29:34.3565577Z TRACE:: DataMonitor.GenerateReport():
DATA USAGE:: Total data requests 1
DATA USAGE:: Succeeded data requests 1
DATA USAGE:: Failed data requests 0
DATA USAGE:: Failed data requests percentage 0%
DATA USAGE:: Total universe data requests 0
DATA USAGE:: Succeeded universe data requests 0
DATA USAGE:: Failed universe data requests 0
DATA USAGE:: Failed universe data requests percentage 0%
2024-02-14T05:29:34.3728470Z TRACE:: Engine.Main(): Packet removed from queue: 1428326774
2024-02-14T05:29:34.3732815Z TRACE:: LeanEngineSystemHandlers.Dispose(): start...
2024-02-14T05:29:34.3739515Z TRACE:: LeanEngineSystemHandlers.Dispose(): Disposed of system handlers.
2024-02-14T05:29:34.3743435Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): start...
2024-02-14T05:29:34.3776890Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): Disposed of algorithm handlers.
